lib/vector-core/src/transform/mod.rs

```diff
@@ -141,6 +141,18 @@ impl Transform {
 ///   `TaskTransform` or vice versa.
 pub trait FunctionTransform: Send + dyn_clone::DynClone + Sync {
     fn transform(&mut self, output: &mut OutputBuffer, event: Event);
+
+    fn transform_all(&mut self, output: &mut OutputBuffer, events: EventArray) {
+        let events_len = events.len();
+        let output_capacity = output.capacity();
+        if events_len > output_capacity {
+            output.reserve(events_len - output_capacity);
+        }
```

Comment:
Do we need to do this calculation? `Vec::reserve` reserves capacity for N more items to be added, not to grow the capacity by N. IOW I think this is sufficient:
```suggestion
        output.reserve(events.len());
```